### ISSUE TITLE
Runs for batchID 128306 are failing to insert in Motus

### ISSUE THREAD

**Issue by denislepage** [jbrzusto/motusServer#374]
runID 25219349 refers to tagID 23201 which does not (no longer) exists in Motus.

I believe this tag was deleted as of a few weeks ago, which suggests that deleted tags from Motus are maintained in sgdata motusServer.

We may need to set a different approach for deleting tags at our end, by flagging them and hiding them from any outputs.

Zoe may know if this tag was deleted as a duplicate of another existing tag.

**Comment by denislepage** [comment]
I have restored, at least temporarily, tag 23201 under project 0, until at least we can resolve this issue properly, to allow the runs data to import.

**Comment by zcrysler** [comment]
Yes, this tag was supposed to be deleted and merged with 23911 as per the call

EXEC prMergeTags 23911 , 23012

There are a number of cases where this was done, including everything in this thread plus some more: https://github.com/MotusDev/ErrorLog/issues/33#issuecomment-365425630

**Comment by jbrzusto** [comment]
I reran a bunch of failed data export jobs.  Normally, the sgdata server refreshes its motus metadata daily, so when a tag is deleted from motus, there might still be detections pushed to the master database from data
processed in the (up to) 24 hours to the next refresh.   These jobs were generally quite a bit older, though.

**Comment by denislepage** [comment]
I was going to update sg_runs, but realized that the merge function below does not refer to 23201. The other github issue doesn’t specify which other tag it was merged with.

I think we are going to keep track of this, maybe in a new table with deleted and merged tag ID’s.

These specific batches were back from December 2017, before we did the merge.

Unless we have kept a list of merges, the only way we can try to figure which tag this run now belongs to is to go back to the backup tables for 23201, and find another tag that matches the same properties that would have been the one retained. In this case, I have already transferred the latest record from the backup table back into the main one.

Unless and until we have a way to modify the sgdata tables,  or deal with the conversions at the time of the import, we need to modify our plans to delete tags from the Motus database.

Let’s wait for Catherine to come back to figure out a path forward. If there are more runs coming through for deleted tags, I will restore all the deleted tags from that list for now. We will need to wait until doing any more deletions too.  I think we have to expect that some outside processes will be referring to older tags, and we need to have ways to deal with that at our end. The 24h lag is probably reasonable, but failed batches that get restarted are more likely to see these problems if the data aren’t reprocessed in full.

I’ll start an issue in motusdev. Maybe this one can be closed in motusServer, as I am not sure there is anything specific that should be done at that end.

**Comment by zcrysler** [comment]
Sorry that was just a typo, these are the other tags that were not included in that other github thread (MotusDev/ErrorLog#33)

EXEC      prMergeTags     23904    ,               23194
EXEC      prMergeTags     23905    ,               23195
EXEC      prMergeTags     23906    ,               23196
EXEC      prMergeTags     23907    ,               23197
EXEC      prMergeTags     23908    ,               23198
EXEC      prMergeTags     23909    ,               23199
EXEC      prMergeTags     23910    ,               23200
EXEC      prMergeTags     23911    ,               23201
EXEC      prMergeTags     23912    ,               23202
EXEC      prMergeTags     23913    ,               23203

**Comment by jbrzusto** [comment]
This issue is a specific case of a more general situation that needs a more general solution:

   https://github.com/jbrzusto/TO_DO/issues/138

When the tag deployment history changes, some receiver data need to be re-run.  There's no
way to get around this for Lotek coded ID tags, where pulses are assembled into bursts
from a tag based on what tags are known to be active.  Pulses assembled into bursts from a tag that turns
out later not to have been active at the time have to be given a new chance to be assigned to bursts from
other tags.  This applies to Lotek receiver data too, because bursts recorded from a tag with Lotek
ID X might be assembled into different runs of that ID if the set of known burst intervals for that ID
active at the time changes.
